### PR TITLE
Disable directory listing for `StaticResource`

### PR DESCRIPTION
**Pull request by dklimpel**
Disable directory listing for `/_matrix/static/`.

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Dirk Klimpel [5740567+dklimpel@users.noreply.github.com](mailto:5740567+dklimpel@users.noreply.github.com)

### CODE REVIEW

File: synapse/http/server.py [hunk 1]
```diff
@@ -569,6 +569,9 @@ def render_GET(self, request: Request) -> bytes:
         set_clickjacking_protection_headers(request)
         return super().render_GET(request)
 
+    def directoryListing(self) -> resource.NoResource:
+        return self.childNotFound
```
Comment (clokep) [inline synapse/http/server.py:573]: I wonder if we should use `twisted.web.pages.notFound` if available due to https://github.com/advisories/GHSA-vg46-2rrj-3647?

(`childNotFound` == `_UnsafeNoResource`)

Comment (dklimpel) [reply synapse/http/server.py]: I did an update to `notFound`.

File: synapse/http/server.py [hunk 2]
```diff
@@ -46,6 +46,8 @@
 from twisted.internet.defer import CancelledError
 from twisted.python import failure
 from twisted.web import resource
+from twisted.web.pages import notFound
```
Comment (clokep) [inline synapse/http/server.py:49]: For old deps I think we need:

```suggestion
try:
    from twisted.web.pages import notFound
except ImportError:
    from twisted.web.resource import NoResource as notFound
```

**Review by clokep — APPROVED** [review]
LGTM If CI is happy.

**Pull request merged**